src/middlewared/middlewared/plugins/vm/pci_linux.py

```diff
@@ -46,30 +46,54 @@ def retrieve_node_information(self, xml):
 
         return info
 
+    async def passthrough_device(self, device):
+        await self.middleware.call('vm.check_setup_libvirt')
+        data = {
+            'capability': {
+                'class': None,
+                'domain': None,
+                'bus': None,
+                'slot': None,
+                'function': None,
+                'product': 'Not Available',
+                'vendor': 'Not Available',
+            },
+            'iommu_group': {},
+            'available': False,
+            'drivers': [],
+            'error': None,
+        }
+        cp = await run(get_virsh_command_args() + ['nodedev-dumpxml', device], check=False)
+        if cp.returncode:
+            data['error'] = cp.stderr.decode()
+            return data
+
+        xml = etree.fromstring(cp.stdout.decode().strip())
+        driver = next((e for e in xml.getchildren() if e.tag == 'driver'), None)
+        drivers = [e.text for e in driver.getchildren()] if driver is not None else []
+
+        node_info = await self.middleware.call('vm.device.retrieve_node_information', xml)
+        return {
+            **node_info, 'drivers': drivers, 'available': node_info['iommu_group']['number'] and all(
```

Comment:
If a dictionary literal is large and has to be multiline, please define its elements each on its own line for readability.